### PR TITLE
Make `min_spanning_tree` robust to incomparable vertex labels

### PR DESCRIPTION
**Pull request by dcoudert**
Part of #35902.

### :books: Description

We ensure that method `min_spanning_tree` operates properly even when vertices and edges are of incomparable types.
We are then able to simplify some code in `src/sage/topology/simplicial_complex.py`.

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [x] I have created tests covering the changes.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: src/sage/topology/simplicial_complex.py [hunk 1]
```diff
@@ -4177,29 +4177,16 @@ def fundamental_group(self, base_point=None, simplify=True):
...
-
         # Edges in the graph may be sorted differently than in the
         # simplicial complex, so convert the edges to frozensets so we
         # don't have to worry about it. Convert spanning_tree to a set
         # to make lookup faster.
-        spanning_tree = set(frozenset((int_to_v[e[0]], int_to_v[e[1]]))
-                            for e in spanning_tree)
+        spanning_tree = set(frozenset((u, v)) for u, v, _ in G.min_spanning_tree())
+        gens = [e for e in G.edge_iterator(labels=False)
+                    if frozenset(e) not in spanning_tree]
```
Comment (fchapoton) [inline src/sage/topology/simplicial_complex.py]: wrong indentation here

**Review by fchapoton — APPROVED** [review]
ok, but the CI is currently broken..

**Review by fchapoton — APPROVED** [review]
tests pass for me in graphs, matroids, topology, so ok

**Comment by dcoudert** [comment]
Thank you.

**Pull request merged**